src/app/GalleryManager.jsx

```diff
@@ -143,7 +145,18 @@ export default class SubmissionGallery extends Component {
   }
 
   confirmEdit(answer) {
-    this.props.dispatch(editAnswer(this.props.forms.editableAnswer, answer, this.props.forms.activeForm));
+    this.props.dispatch(editAnswer(this.props.forms.editableAnswer, answer.submission_id, answer.answer_id, this.props.forms.activeForm));
+
+    // probably a better way to do this.
+    // UI is going to get weird if there are lots of PII answers
+
+    // filter out answers that have not changed
+    this.props.forms.editablePii
+      // this is going to re-edit an edited question, but oh well.
+      .filter(idAns => idAns.edited)
+      .forEach(idAns => {
+        this.props.dispatch(editAnswer(idAns.edited, answer.submission_id, idAns.widget_id, this.props.forms.activeForm));
```

Comment:
This is great.  This is the exact kind of simplicity that we win by treating PII as just another question.  The complexity, of course, it pushed to the UX... but, hey, this is simple and completely flexible.